src/app/features/courses/services/courses.service.ts

```diff
@@ -0,0 +1,125 @@
+      imagePath: '../../../../assets/reactjs.png',
+      topRated: false
+    },
+    {
+      id: 3,
+      title: 'Vue',
+      creationDate: +new Date(2019, 7, 24),
+      duration: 90,
+      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
+      imagePath: '../../../../assets/vuejs.jpg',
+      topRated: false
+    },
+    {
+      id: 4,
+      title: 'Git',
+      creationDate: +new Date(2019, 5, 30),
+      duration: 54,
+      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
+      imagePath: '../../../../assets/git_new.png',
+      topRated: false
+    },
+    {
+      id: 5,
+      title: 'Javascript',
+      creationDate: +new Date(2019, 1, 18),
+      duration: 200,
+      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
+      imagePath: '../../../../assets/javascript.png',
+      topRated: false
+    },
+    {
+      id: 6,
+      title: 'Jquery',
+      creationDate: +new Date(2019, 2, 2),
+      duration: 100,
+      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
+      imagePath: '../../../../assets/jquery_new.png',
+      topRated: false
+    },
+    {
+      id: 7,
+      title: 'Ember.js',
+      creationDate: +new Date(2019, 3, 3),
+      duration: 333,
+      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
+      imagePath: '../../../../assets/ember.jpg',
+      topRated: false
+    },
+    {
+      id: 8,
+      title: 'Express',
+      creationDate: +new Date(2019, 4, 4),
+      duration: 123,
+      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
+      imagePath: '../../../../assets/express_new.png',
+      topRated: false
+    },
+    {
+      id: 9,
+      title: 'Node.js',
+      creationDate: +new Date(2019, 5, 5),
+      duration: 950,
+      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
+      imagePath: '../../../../assets/node_new.jpeg',
+      topRated: false
+    }
+  ];
+
+  getList(): CoursesItemModel[] {
+    return this.coursesList;
+  }
+
+  createCourse(course: CoursesItemModel): void {
+    this.coursesList = [...this.coursesList, course];
+  }
+
+  getCourse(id: number): CoursesItemModel {
+    return this.coursesList.filter( course => course.id === id)[0];
+  }
+
+  updateCourse(updatedCourse: CoursesItemModel): void {
+    this.coursesList.map( (course, index) => {
+      if (course.id === updatedCourse.id) {
+        this.coursesList = [
+          ...this.coursesList.slice(0, index),
+          updatedCourse,
+          ...this.coursesList.slice(index + 1)
+        ];
+      }
+    });
+  }
+
+  remove(courseToRemove: CoursesItemModel): void {
+    this.coursesList.map( (course, index) => {
+      if (course.id === courseToRemove.id) {
+        this.coursesList = [
+          ...this.coursesList.slice(0, index),
+          ...this.coursesList.slice(index + 1)
+        ];
+      }
+    });
```

Comment:
minor: 
this.coursesList = this.coursesList.reduce((allCurses, course) => {
if(course.id !== courseToRemove.id) {
allCurses.push(course);
}
return allCurses;
}, [] )

Reply:
cool, thanks for the advice.